### PR TITLE
Update remote OVAL location for RHEL products

### PR DESCRIPTION
**Pull request by ggbecker**
#### Description:

- Update the location of remote OVAL files for rule `security_patches_up_to_date`.

Note:  There are only compressed files in this new location path but OpenSCAP is able to handle them.

#### Rationale:

- Old location is now deprecated per message on: https://www.redhat.com/security/data/oval/

```
NOTE: the files listed here are Deprecated!

Instead use the product-specific OVAL Stream or SCAP source data stream files.
```

### CODE REVIEW

**Comment by yuumasato** [comment]
> Note: There are only compressed files in this new location path but OpenSCAP is able to handle them.

I believe this will affect SCAP-1.3.2 validation of SCAP 1.3 content.

**Comment by jan-cerny** [comment]
@yuumasato Does it reject bzip files?

**Comment by jan-cerny** [comment]
@openscap-ci test this please

**Comment by shawndwells** [comment]
@yuumasato isn't SCAL validation part of the test suite?

**Comment by jan-cerny** [comment]
Partially yes. The XML schema validation is part of the PR gating. But the SCAPVal tool by NIST performs many additional checks, it takes a long time, so it's executed nightly: https://jenkins.complianceascode.io/job/scap-security-guide-scapval-scap-1.3/.

**Comment by ggbecker** [comment]
> > Note: There are only compressed files in this new location path but OpenSCAP is able to handle them.
> 
> I believe this will affect SCAPVAL-1.3.2 validation of SCAP 1.3 content.
I have executed some tests and

SCAPVAL-1.3.2 doesn't work with `bzip` files:
```
2020-04-30 17:27:50 INFO  Validating SCAP Content: ssg-rhel8-ds.xml (SHA-256: 9CD506BFD694701A557D4481DFFF22C752BD5E200FC92795FCD1FD7F89C75EC0) with SCAP Version: 1.3
2020-04-30 17:27:50 INFO  Found a remote component-ref with id scap_org.open-scap_cref_security-data-oval-v2-RHEL8-rhel-8.oval.xml.bz2 and URL https://www.redhat.com/security/data/oval/v2/RHEL8/rhel-8.oval.xml.bz2 will attempt to acquire and include in validation. 
2020-04-30 17:27:50 INFO  Downloading: https://www.redhat.com/security/data/oval/v2/RHEL8/rhel-8.oval.xml.bz2 ...
2020-04-30 17:27:52 INFO  Unable to utilize remote component-ref with id scap_org.open-scap_cref_security-data-oval-v2-RHEL8-rhel-8.oval.xml.bz2 and url https://www.redhat.com/security/data/oval/v2/RHEL8/rhel-8.oval.xml.bz2 - Error on line 1 of document file:/tmp/1588260471803390940775966921743.tmp: Content is not allowed in prolog.
```

Regular XML file:
```
2020-04-30 17:13:56 INFO  Validating SCAP Content: ssg-rhel8-ds.xml (SHA-256: F183A07AF3A49D3FF62E8C1E261A2776C4EB3ED572DAD90B9E12FA18AF06553E) with SCAP Version: 1.3
2020-04-30 17:13:57 INFO  Found a remote component-ref with id scap_org.open-scap_cref_security-data-oval-com.redhat.rhsa-RHEL8.xml and URL https://www.redhat.com/security/data/oval/com.redhat.rhsa-RHEL8.xml will attempt to acquire and include in validation. 
2020-04-30 17:13:57 INFO  Downloading: https://www.redhat.com/security/data/oval/com.redhat.rhsa-RHEL8.xml ...
```

We should also extend [tests/run_scapval.py](https://github.com/ComplianceAsCode/content/blob/master/tests/run_scapval.py#L85) to use the option `-online` so it downloads latest CPE's and fetch remote resources such as the OVAL file mentioned in this pull request.

```
 -online                 Enable download of latest dictionaries and remote
                         resolution of some components
```

**Comment by yuumasato** [comment]
Between fetching deprecated CVE feed, and failing SCAPVAL-1.3.2, I'd go with failing SCAPVAL.

**Comment by jan-cerny** [comment]
If you disable SCAPVAL, how will we test that our content will  be accepted by NIST?

**Comment by ggbecker** [comment]
> If you disable SCAPVAL, how will we test that our content will be accepted by NIST?

That's a great question. Maybe we should check with @isimluk if he has an opinion on this and why are only compressed OVAL files available for the version 2 of this repository (https://www.redhat.com/security/data/oval/v2/).

**Comment by matejak** [comment]
TLDR: The future is compressed, it is extremely unlikely that uncompressed feeds are coming back, so we should probably reach to scapval maintainers, so they enhance the tool.

The full story: Downloads of the OVAL feed generates an unimaginably huge amount of traffic, and the infrastructure that is serving that data is itself obsolete and for some reasons hard to upgrade. As there are more and more feeds and existing feeds get longer and longer, space-saving features need to be enabled both on the storage side and the transport side. Zipping feeds has therefore a good ratio of price/value, and is not likely to go away.

**Comment by matejak** [comment]
We almost certainly don't have to change this for RHEL<8. The solution for RHEL8 and future RHELs is still unknown, we hope for uncompressed feed taken from the `v2` directory, but it has yet to be confirmed.

**Comment by ggbecker** [comment]
Note: latest scapval (1.3.5) still fails on the remote resource requirement.

**Comment by redhatrises** [comment]
> Note: latest scapval (1.3.5) still fails on the remote resource requirement.

This isn't going to change anytime soon until a bump of the SCAP specification. We will have to bite the bullet right now for any and all SCAP 1.3 content.

**Comment by redhatrises** [comment]
Closing this hasn't been worked on in a long while.

**Comment by matejak** [comment]
Summary: This hasn't been worked on is that the compressed feed breaks SCAP validation, and in the meantime, the OpenSCAP scanner got support for HTTP-level transfer compression of uncompressed contents. As a result, we expect to be able to stay with uncompressed feeds for a longer time i.e. the problem this PR aimed to solve isn't urgent.
